examples/templates/configuration/raw-keyrings/raw-rsa-keyring-encrypt-with-public-only.md

```diff
@@ -0,0 +1,188 @@
+
+Implementations of this example MUST follow the rules defined in
+[Example Templates](../../../examples.md#example-templates).
+
+## Implementations
+
+- [Python (DEV)](https://github.com/aws/aws-encryption-sdk-python/blob/keyring/examples/src/keyring/raw_rsa/public_private_key_separate.py)
+- [Java (DEV)](https://github.com/aws/aws-encryption-sdk-java/blob/keyring/src/examples/java/com/amazonaws/crypto/examples/keyring/rawrsa/PublicPrivateKeySeparate.java)
+
+## Definitions
+
+### Conventions used in this document
+
+The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in
+[RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Header
+
+```python
+# One of the benefits of asymmetric encryption
+# is that you can encrypt with just the public key.
+# This means that you can give someone the ability to encrypt
+# without giving them the ability to decrypt.
+#
+# The raw RSA keyring supports encrypt-only operations
+# when it only has access to a public key.
+#
+# This example shows how to construct and use the raw RSA keyring
+# to encrypt with only the public key and decrypt with the private key.
+#
+# If your RSA key is in PEM or DER format,
+# see the ``keyring/raw_rsa/private_key_only_from_pem`` example.
+#
+# https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/choose-keyring.html#use-raw-rsa-keyring
+#
+# In this example, we use the one-step encrypt and decrypt APIs.
+```
+
+## Summary
+
+```python
+# Demonstrate an encrypt/decrypt cycle using separate public and private raw RSA keyrings.
+```
+
+## Inputs
+
+- **source plaintext** :
+  Plaintext to encrypt
+
+## Steps
+
+1. Define encryption context.
+
+   ```python
+   # Prepare your encryption context.
+   # Remember that your encryption context is NOT SECRET.
+   # https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/concepts.html#encryption-context
+   encryption_context = {
+       "encryption": "context",
+       "is not": "secret",
+       "but adds": "useful metadata",
+       "that can help you": "be confident that",
+       "the data you are handling": "is what you think it is",
+   }
+   ```
+
+1. Generate the wrapping key.
+
+   ```python
+   # Generate an RSA private key to use with your keyring.
+   # In practice, you should get this key from a secure key management system such as an HSM.
+   #
+   # The National Institute of Standards and Technology (NIST) recommends a minimum of 2048-bit keys for RSA.
+   # https://www.nist.gov/publications/transitioning-use-cryptographic-algorithms-and-key-lengths
+   #
+   # Why did we use this public exponent?
+   # https://crypto.stanford.edu/~dabo/pubs/papers/RSA-survey.pdf
+   private_key = rsa.generate_private_key(public_exponent=65537, key_size=4096, backend=default_backend())
+   ```
+
+1. Derive the public key from the wrapping private key.
```

Comment:
This feels like something we should include or put elsewhere.
So that it is super clear.
And have the function just return the pairs...
?